src/Symfony/Component/Serializer/DataCollector/SerializerDataCollector.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\DataCollector;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\Serializer\Debug\Normalizer\AbstractTraceableNormalizer;
+use Symfony\Component\Serializer\Debug\TraceableSerializer;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\VarDumper\Cloner\Data;
+use Symfony\Contracts\Service\ResetInterface;
+use Throwable;
+
+class SerializerDataCollector extends DataCollector
+{
+    /**
+     * @var SerializerInterface|null
+     */
+    private $serializer;
+    /**
+     * @var array|iterable
+     */
+    private $normalizers;
```

Comment:
```suggestion
    private $serializer;
    private $normalizers;
```

Those types can be inferred from the constructor. We don't add a doc block in those cases.